lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java

```diff
@@ -318,6 +321,29 @@ public ValueDecoder<long[]> getTimeMicrosDecoder(ParquetEncoding encoding)
                 });
     }
 
+    public ValueDecoder<long[]> getTimeMillisDecoder(ParquetEncoding encoding)
+    {
+        int precision = ((TimeType) field.getType()).getPrecision();
+        if (precision < 3) {
+            return new InlineTransformDecoder<>(
+                    getInt32ToLongDecoder(encoding),
+                    (values, offset, length) -> {
+                        // decoded values are millis, round to lower precision and convert to picos
+                        // modulo PICOSECONDS_PER_DAY is applied for the case when a value is rounded up to PICOSECONDS_PER_DAY
```

Comment:
should we have similar rounding for `getTimeMicrosDecoder` where trino type is lower than precision than 6?
(or assertion that type is 6 if only time(6) is supported)

Reply:
Yes, I'm planning to tackle that separately by adding rounding on lower precision. In practise we're only supporting time(6) in iceberg right now, so it's not a problem yet. But from parquet reader perspective it is better to assume such a scenario might arise in future and support it correctly.